src/Psalm/Internal/Provider/ReturnTypeProvider/PdoStatementReturnTypeProvider.php

```diff
@@ -17,19 +15,20 @@ public static function getClassLikeNames(): array
 
     public static function getMethodReturnType(MethodReturnTypeProviderEvent $event): ?Type\Union
     {
+        $config = Config::getInstance();
         $source = $event->getSource();
         $call_args = $event->getCallArgs();
         $method_name_lowercase = $event->getMethodNameLowercase();
         if ($method_name_lowercase === 'fetch'
-            && class_exists('PDO')
+            && $config->php_extensions["pdo"]
             && isset($call_args[0])
             && ($first_arg_type = $source->getNodeTypeProvider()->getType($call_args[0]->value))
             && $first_arg_type->isSingleIntLiteral()
         ) {
             $fetch_mode = $first_arg_type->getSingleIntLiteral()->value;
 
             switch ($fetch_mode) {
-                case PDO::FETCH_ASSOC: // array<string,scalar|null>|false
+                case 2: // PDO::FETCH_ASSOC - array<string,scalar|null>|false
```

Comment:
I don't really like using literals instead of consts, but I think it's worth it to allow psalm to be used for the PDO extension without having the PDO extension loaded at runtime.

Reply:
Yeah, we sometimes have to do it anyway when constant value change between php versions anyway